src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig

```diff
@@ -308,10 +308,12 @@ file that was distributed with this source code.
                         {% endif %}
 
                         // append to Select2
-                        autocompleteInput.attr('data', data).append(data).trigger('change');
+                        // NEXT_MAJOR: Remove ".select2('data', data)" while dropping v3 support
+                        autocompleteInput.select2('data', data).append(data).trigger('change');
 
                         // manually trigger the `select2:select` event
-                        autocompleteInput.attr('data', data).trigger({
+                        // NEXT_MAJOR: Remove ".select2('data', data)" while dropping v3 support
```

Comment:
```suggestion
                        // NEXT_MAJOR: Change to ".attr('data', data)" while dropping v3 support
```

Reply:
```
Can you try this ? @alistairstead @willemverspyck

// append to Select2
autocompleteInput.append(data).trigger('change');

// manually trigger the `select2:select` event
autocompleteInput.trigger({
    type: 'select2:select',
    params: {
        data: data
    }
});
This changes works for me too (master) and its better code. @VincentLanglet
```

As mentioned by @cecicifu in #7210, we can remove ".select2('data', data)". I also tested this in the master branch and that works. Same for line 311.

Reply:
Ok thanks

Reply:
The improvement i have made is only for master branch, need to open a new PR in master branch, right ?

Reply:
I will ping you when I'll do the merge of 3.x into master after this merge @cecicifu